### PR TITLE
Move all HTTP specific code to Api\Http layer in preparation for PB Support

### PR DESCRIPTION
**Pull request by christophermancini**
In order to support PB, needed to decouple http code from the core lib and move it all to the API layer. Since this is a major change and potentially has breaking changes in it, this will be a brand new minor release for the PHP lib.

### CODE REVIEW

**Comment by debo** [comment]
@christophermancini I think might even be a Major if you introduce BC

**Comment by christophermancini** [comment]
So, the only things that changed interface wise, are headers are no longer relevant. So, instead of setting the Vclock header on the command builder or as a header on the object, you set it on the object. A few other small changes like that. The other major changes were with response objects, they no longer have anything to do with headers, they have a success flag, a code (http status code on http side), a message and then any command relevant properties.

I am not sure its enough of a change to warrant a v3?

**Comment by debo** [comment]
@christophermancini if you were to be strict about semver if there is BC then you bump the Major, even if the change is small. The last call is yours though.

**Comment by christophermancini** [comment]
@debo Fair enough, lets be strict about semver, this way no one can criticize us for it later. :)

**Comment by debo** [comment]
@christophermancini as they say, better be wrong then sorry ;)

**Comment by lukebakken** [comment]
@christophermancini build failures ... looks like something to do with code coverage?

**Comment by christophermancini** [comment]
@lukebakken not sure why it failed only for php5.6, I reran it and it worked fine. :question:

**Comment by lukebakken** [comment]
That is really strange that a re-run would fix it.

File: tests/unit/Riak/ObjectTest.php [hunk 1]
```diff
@@ -112,7 +112,7 @@ public function testGetIndex()
     }
 
     /**
-     * @expectedException InvalidArgumentException
+     * @expectedException \InvalidArgumentException
```
Comment (lukebakken) [inline tests/unit/Riak/ObjectTest.php]: Stray `\` ?

**Comment by lukebakken** [comment]
Looks fine to me :+1:

**Comment by mjbrender** [comment]
Hey @debo -- you've been incredibly helpful here! I'd love to thank you by sending some Basho stickers. Shoot me an email to mbrender /at/ basho.com!

**Pull request merged**